features/roomdetails/api/src/main/kotlin/io/element/android/features/roomdetails/api/RoomDetailsEntryPoint.kt

```diff
@@ -16,11 +16,32 @@
 
 package io.element.android.features.roomdetails.api
 
+import android.os.Parcelable
 import com.bumble.appyx.core.modality.BuildContext
 import com.bumble.appyx.core.node.Node
 import com.bumble.appyx.core.plugin.Plugin
 import io.element.android.libraries.architecture.FeatureEntryPoint
+import io.element.android.libraries.architecture.NodeInputs
+import io.element.android.libraries.matrix.api.core.UserId
+import kotlinx.parcelize.Parcelize
 
 interface RoomDetailsEntryPoint : FeatureEntryPoint {
-    fun createNode(parentNode: Node, buildContext: BuildContext, plugins: List<Plugin>): Node
+
+    sealed interface NavTarget : Parcelable {
+        @Parcelize
+        object RoomDetails : NavTarget
+
+        @Parcelize
+        object RoomMemberList : NavTarget
+
+        @Parcelize
+        object InviteMembers : NavTarget
+
+        @Parcelize
+        data class RoomMemberDetails(val roomMemberId: UserId) : NavTarget
+    }
```

Comment:
I understand that we do not want to duplicate `NavTarget`, but this is a bit strange to expose all the targets to the API module. Some targets may stay internal.

I would create a new type here:

```kotlin
    sealed interface InitialTarget {
        // Default to RoomDetails
        object Default : InitialTarget

        data class RoomMemberDetails(val roomMemberId: UserId) : InitialTarget
    }
```

and in the impl module, a conversion fun:

```kotlin
private fun RoomDetailsEntryPoint.InitialTarget.toInternalTarget(): RoomDetailsFlowNode.NavTarget {
    return when (this) {
        RoomDetailsEntryPoint.InitialTarget.Default -> RoomDetailsFlowNode.NavTarget.RoomDetails
        is RoomDetailsEntryPoint.InitialTarget.RoomMemberDetails -> RoomDetailsFlowNode.NavTarget.RoomMemberDetails(roomMemberId)
    }
}
```

Not sure if it worth to change it now though.

Reply:
Yes I'd have done it like that too

Reply:
Yes, I moved stuff around just to check if it worked, but I forgot to do a proper mapping here. I'll move it back to impl and add the `InitialTarget` here.

Reply:
Done in https://github.com/vector-im/element-x-android/pull/482/commits/42dd2726899a9835915f9302e3ece2ce94477589